Src/zipkin4net/Src/ISpanCustomizer.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+
+namespace zipkin4net
+{
+    /// <summary>
+    /// Simple interface users can customize a span with. For example, this can add custom tags useful
+    /// in looking up spans.
+    /// 
+    /// <p>This type is safer to expose directly to users than {@link Span}, as it has no hooks that
+    /// can affect the span lifecycle.</p>
+    /// </summary>
+    //TODO internal for now. It's still work in progress
+    internal interface ISpanCustomizer
+    {
+        /// <summary>
+        /// Sets the string name for the logical operation this span represents.
+        /// </summary>
+        ISpan Name(string name);
+        
+        /// <summary>
+        /// Associates an event that explains latency with the current system time.
+        /// </summary>
+        /// <param name="value">A short tag indicating the event, like "finagle.retry"</param>
+        ISpan Annotate(string value);
+        
+        /// <summary>
+        /// Like <see cref="Annotate(string)"/>, except with a given timestamp in microseconds.
+        /// </summary>
+        ISpan Annotate(DateTime timestamp, string value);
```

Comment:
same commentary about pre-ordained here, though interfaces might be trickier as harder to add methods to later (at least in old java not sure in dotnet)